### DISCUSSION TITLE
Import code from online source

### DISCUSSION THREAD

**Discussion by ghost** [vlang/v#16942] [Ideas]
Hi there. I just want to suggest a feature to the ``import`` type. You can import code into code by providing URL to that code
E.g.: ``import https://example.com/code.v``

**Comment by MatejMagat305** [comment]
I do not want to attack, only ask, this is "temp" of command "`v install package`" - would be better to some "`v tidy .`" and `import example.com/...` similar  `go mod tidy`? But it is planing or implemented, is not it?

**Comment by JalonSolov** [comment]
While it could be "handy" in some cases, it also open a lot of security problems... such as someone downloading a random bit of V code and trying to compile it, and suddenly they've imported malware without realizing.

The plan is for V to automatically do `v install` for you on missing imports, but those imports still need to be registered with `vpm`.

**Comment by ghost** [comment]
> Hi there. I just want to suggest a feature to the ``import`` type. You can import code into code by providing URL to that code
> E.g.: ``import https://example.com/code.v``

Took inspiration from Go, where you can already do that.

**Reply by JalonSolov** [reply]
V tries to be safe, and that isn't.  :-\

**Reply by MatejMagat305** [reply]
> > Hi there. I just want to suggest a feature to the `import` type. You can import code into code by providing URL to that code
> > E.g.: `import https://example.com/code.v`
> 
> Took inspiration from Go, where you can already do that.

neigher golang is not so easy golang is download it from github similar to vlang and not one file, but whole package ..., so I do not autor of vlang but I do not understand you, what you sugest?

**Reply by Wajinn** [reply]
> > > E.g.: `import https://example.com/code.v`
> > 
> > 
> > Took inspiration from Go, where you can already do that.
> 
> neigher golang is not so easy golang is download it from github similar to vlang and not one file, but whole package ..., so I do not autor of vlang but I do not understand you, what you sugest?

Vlang has [VPM](https://github.com/vlang/vpm).  Thus the modules were looked at and approved, versus creating a security risk situation where things are randomly downloaded from anywhere.
